### ISSUE TITLE
How do I respond with only meta data?

### ISSUE THREAD

**Issue by ezekg** [rails-api/active_model_serializers#1795]
I am using the `:json_api` adapter and am wanting to render only meta data for a few endpoints, e.g. responding without a top-level `data` key. As far as I can tell, [this is allowable under the JSON API spec](http://jsonapi.org/format/#document-top-level). I would expect to be able to do something similar to,

```ruby
render: meta: { ... }, serializer: SomeMetaSerializer
```

But even passing in an empty (or `nil`) `json: {}` hash throws errors like `undefined method 'read_attribute_for_serialization' for nil:NilClass`. I know I could simply do,

```ruby
render: json: { meta: { ... } }.to_json
```

but I'd like to take advantage of active model serializers automatic key transform.

Is there a way to do this that I'm just missing? Am I doing it wrong? I've attempted a few different monkey patches, but there doesn't seem to be a way to get rid of the top-level `data` key, even though it seems it's not explicitly required if a `meta` or `errors` key is present.

If needed, I am using Rails `5.0.0.rc1`. with AMS `0.10.0`.

**Comment by bf4** [comment]
I do this in my app a lot. But I just generate my JSON directly when using meta:

```ruby
render_options = {}
render_options[:status] = :ok
render_options[:location] = nil
render_options[:jsonapi] = { meta: { user_id: user_id } }.to_json
render render_options
```

You can use the TransformKeys utility function directly:

```ruby
ActiveModelSerializers::KeyTransform.camel({meta: { user_id: 5} })
=> {:Meta=>{:UserId=>5}}
```

But looks like this works, though I think the above is better:

```ruby
ActiveModelSerializers::SerializableResource.new([], {meta: { user_id: 5 } }).as_json.except(nil)
# => { "meta" => {  user_id: 5 } }
```

**Comment by bf4** [comment]
If this works, please share back and improve the docs!

**Comment by ezekg** [comment]
I'm currently doing something similar but got a little frustrated with the repetition. Shouldn't we be able to render `json: nil` if a `meta` option is passed? In any case, I was able to get what I wanted by defining a helper in my `ApplicationController`,

```ruby
def render_meta(meta)
  render json: ActiveModelSerializers::KeyTransform.camel_lower(meta: meta).to_json
end
```

**Comment by bf4** [comment]
`json: nil` makes less sense than `json: {}` in this scenario, since `json:` refers to the JSON renderer. I'm not sure how we should should handle this situation as AMS, but what you've done and what I've done works for me :)

**Comment by ezekg** [comment]
Fair enough. I'd be happy to add something in the docs about this. Any place in particular I should add that in, just another section under the `docs/howo` directory?

**Comment by remear** [comment]
@ezekg Sure, that sounds like a good place to start.
